src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/RequestSqlController.php

```diff
@@ -0,0 +1,449 @@
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Responsible of "Configure > Advanced Parameters > Database -> SQL Manager" page
+ */
+class RequestSqlController extends FrameworkBundleAdminController
+{
+    /**
+     * Show list of saved SQL's
+     *
+     * @AdminSecurity(
+     *     "is_granted(['read','update', 'create','delete'], request.get('_legacy_controller')~'_')",
+     *      message="Access denied."
+     * )
+     *
+     * @param Request           $request
+     * @param RequestSqlFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, RequestSqlFilters $filters)
+    {
+        // handle "Export to SQL manager" action from legacy pages
+        if ($request->query->has('addrequest_sql')) {
+            return $this->forward('PrestaShopBundle:Admin\Configure\AdvancedParameters\RequestSql:create');
+        }
+
+        $gridLogFactory = $this->get('prestashop.core.grid.factory.request_sql');
+        $grid = $gridLogFactory->createUsingSearchCriteria($filters);
+
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+        $presentedGrid = $gridPresenter->present($grid);
+
+        $settingsForm = $this->getSettingsFormHandler()->getForm();
+
+        return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/RequestSql/list.html.twig', [
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->generateUrl('admin_request_sql_create'),
+                    'desc' => $this->trans('Add new SQL query', 'Admin.Advparameters.Feature'),
+                    'icon' => 'add_circle_outline',
+                ],
+            ],
+            'layoutTitle' => $this->trans('SQL Manager', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'requestSqlSettingsForm' => $settingsForm->createView(),
+            'requestSqlGrid' => $presentedGrid,
+        ]);
+    }
+
+    /**
+     * @AdminSecurity("is_granted(['read', 'update', 'create', 'delete'], request.get('_legacy_controller')~'_')", message="You do not have permission to update this.", redirectRoute="admin_logs")
+     * @DemoRestricted(redirectRoute="admin_logs")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        if (!$request->request->has('request_sql')) {
+            return $this->redirectToRoute('admin_request_sql');
+        }
+
+        $filters = $request->request->get('request_sql');
+
+        return $this->redirectToRoute('admin_request_sql', ['filters' => [
+            'id_request_sql' => $filters['id_request_sql'],
+            'name' => $filters['name'],
+            'sql' => $filters['sql'],
+        ]]);
```

Comment:
@mickaelandrieu this is temporary solution untill we can get form type from grid. :/

Reply:
ok :+1:  can you create a minimal issue on prestafony project so we won't forget it? I'm more a gold fish than a human to remember this kind of stuff

Reply:
nooope, this can actually be updated with dynamic form now, ill do that. :+1: